Humble implicit conversion warnings

Compilation in humble gives such warnings:
```
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h: In function ‘void pcl_conversions::fromPCL(const uint64_t&, rclcpp::Time&)’:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:84:36: error: conversion to ‘int64_t’ {aka ‘long int’} from ‘long long unsigned int’ may change the sign of the result [-Werror=sign-conversion]
   84 |     stamp = rclcpp::Time(pcl_stamp * 1000ull); // Convert from us to ns
      |                          ~~~~~~~~~~^~~~~~~~~
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h: In function ‘void pcl_conversions::toPCL(const rclcpp::Time&, uint64_t&)’:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:90:34: error: conversion to ‘long long unsigned int’ from ‘rcl_time_point_value_t’ {aka ‘long int’} may change the sign of the result [-Werror=sign-conversion]
   90 |     pcl_stamp = stamp.nanoseconds() / 1000ull;  // Convert from ns to us
      |                 ~~~~~~~~~~~~~~~~~^~
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h: In function ‘void pcl_conversions::fromPCL(const std::vector<pcl::PCLPointField>&, std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >&)’:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:216:26: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  216 |       fromPCL(*(it), pfs[i]);
      |                          ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h: In function ‘void pcl_conversions::toPCL(const std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >&, std::vector<pcl::PCLPointField>&)’:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:236:28: error: conversion to ‘std::vector<pcl::PCLPointField>::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  236 |       toPCL(*(it), pcl_pfs[i]);
      |                            ^
In file included from /as_drive/ws/src/as_drive_nodes/visualization/as_lidar_semantic_segmentation_viz/test/lidar_semantic_segmentation_viz_test.cpp:19:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h: In function ‘bool pcl::concatenatePointCloud(const PointCloud2&, const PointCloud2&, sensor_msgs::msg::PointCloud2&)’:
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:686:52: error: conversion to ‘unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  686 |                                  pcl::getFieldSize (cloud2.fields[j].datatype));
      |                                  ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:685:57: error: conversion to ‘std::vector<int>::value_type’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Werror=sign-conversion]
  685 |         fields2_sizes.push_back (cloud2.fields[j].count *
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:698:29: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  698 |           if (cloud1.fields[i].name == "_")
      |                             ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:705:30: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  705 |           if ((cloud1.fields[i].name == "rgb" && fields2[j].name == "rgba") ||
      |                              ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:706:30: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  706 |               (cloud1.fields[i].name == "rgba" && fields2[j].name == "rgb") ||
      |                              ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:707:30: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  707 |               (cloud1.fields[i].name == fields2[j].name))
      |                              ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:709:109: error: conversion to ‘std::vector<sensor_msgs::msg::PointField_<std::allocator<void> >, std::allocator<sensor_msgs::msg::PointField_<std::allocator<void> > > >::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  709 |             memcpy (reinterpret_cast<char*> (&cloud_out.data[nrpts + cp * cloud1.point_step + cloud1.fields[i].offset]),
      |                                                                                                             ^
/opt/ros/humble/include/pcl_conversions/pcl_conversions.h:709:20: error: conversion to ‘size_t’ {aka ‘long unsigned int’} from ‘__gnu_cxx::__alloc_traits<std::allocator<int>, int>::value_type’ {aka ‘int’} may change the sign of the result [-Werror=sign-conversion]
  709 |             memcpy (reinterpret_cast<char*> (&cloud_out.data[nrpts + cp * cloud1.point_step + cloud1.fields[i].offset]),
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  710 |                     reinterpret_cast<const char*> (&cloud2.data[cp * cloud2.point_step + cloud2.fields[j].offset]),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  711 |                     fields2_sizes[j]);
      |                     ~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
````

Compilator version
```
gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/pcl_conversions/include/pcl_conversions/pcl_conversions.h
+++ b/pcl_conversions/include/pcl_conversions/pcl_conversions.h
@@ -37,6 +37,7 @@
 #ifndef PCL_CONVERSIONS_H__
 #define PCL_CONVERSIONS_H__
 
+#include <cstddef>
 #include <vector>
 
 #include <rclcpp/rclcpp.hpp>
@@ -81,13 +82,14 @@ namespace pcl_conversions {
   inline
   void fromPCL(const std::uint64_t &pcl_stamp, rclcpp::Time &stamp)
   {
-    stamp = rclcpp::Time(pcl_stamp * 1000ull); // Convert from us to ns
+    stamp = rclcpp::Time(
+      static_cast<rcl_time_point_value_t>(pcl_stamp * 1000ull));  // Convert from us to ns
   }
 
   inline
   void toPCL(const rclcpp::Time &stamp, std::uint64_t &pcl_stamp)
   {
-    pcl_stamp = stamp.nanoseconds() / 1000ull;  // Convert from ns to us
+    pcl_stamp = static_cast<std::uint64_t>(stamp.nanoseconds()) / 1000ull;  // Convert from ns to us
   }
 
   inline
@@ -211,7 +213,7 @@ namespace pcl_conversions {
   {
     pfs.resize(pcl_pfs.size());
     std::vector<pcl::PCLPointField>::const_iterator it = pcl_pfs.begin();
-    int i = 0;
+    size_t i = 0;
     for(; it != pcl_pfs.end(); ++it, ++i) {
       fromPCL(*(it), pfs[i]);
     }
@@ -231,7 +233,7 @@ namespace pcl_conversions {
   {
     pcl_pfs.resize(pfs.size());
     std::vector<sensor_msgs::msg::PointField>::const_iterator it = pfs.begin();
-    int i = 0;
+    size_t i = 0;
     for(; it != pfs.end(); ++it, ++i) {
       toPCL(*(it), pcl_pfs[i]);
     }
@@ -676,15 +678,16 @@ namespace pcl {
     {
       // Get the field sizes for the second cloud
       std::vector<sensor_msgs::msg::PointField> fields2;
-      std::vector<int> fields2_sizes;
+      std::vector<size_t> fields2_sizes;
       for (size_t j = 0; j < cloud2.fields.size (); ++j)
       {
         if (cloud2.fields[j].name == "_")
           continue;
 
-        fields2_sizes.push_back (cloud2.fields[j].count *
-                                 pcl::getFieldSize (cloud2.fields[j].datatype));
-        fields2.push_back (cloud2.fields[j]);
+        fields2_sizes.push_back(
+          cloud2.fields[j].count *
+          static_cast<size_t>(pcl::getFieldSize(cloud2.fields[j].datatype)));
+        fields2.push_back(cloud2.fields[j]);
       }
 
       cloud_out.data.resize (nrpts + (cloud2.width * cloud2.height) * cloud_out.point_step);
@@ -692,7 +695,7 @@ namespace pcl {
       // Copy the second cloud
       for (size_t cp = 0; cp < cloud2.width * cloud2.height; ++cp)
       {
-        int i = 0;
+        size_t i = 0;
         for (size_t j = 0; j < fields2.size (); ++j)
         {
           if (cloud1.fields[i].name == "_")
@@ -737,7 +740,7 @@ namespace pcl {
 
 } // namespace pcl
 
-/* TODO when ROS2 type masquareding is implemented */ 
+/* TODO when ROS2 type masquerading is implemented */ 
 /**
 namespace ros
 {

